File: solutions/0003-longest-substring-without-repeating-characters/solution.py
class Solution:
	def lengthOfLongestSubstring(self, s: str) -> int:
		start = -1    # start index of sliding window
		max_length = 0       # the longest substring length
		d = {}        # store non-repeating characters in sliding window 

		for i in range(len(s)):     

			if s[i] in d and d[s[i]] > start:
				start = d[s[i]]
				d[s[i]] = i

			else: #s[i] not in d
				d[s[i]] = i

				if i - start > max_length:  # (i - start) is the length of sliding window  
					max_length = i-start

		return max_length

File: solutions/0003-longest-substring-without-repeating-characters/solution.c
#include <stdio.h>
#include <string.h>

int lengthOfLongestSubstring(char *s) {


int n = strlen(s);
    int start = 0, end = 0;
    int maxLen = 0;
     int set[128] = {0};  // initialize set with 0

    while (end < n) {
        if (set[s[end]] == 0) {

            set[s[end]] = 1;
              end++;
            maxLen =    (end - start > maxLen) ? (end - start) : maxLen;
        } else {

            set[s[start]] = 0;
            start++;
        }
    }

    return maxLen;
}



